Design guidelines: Drawer and splitter (#5088)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/components/drawer/drawer.md

### Splitter in a drawer

> A splitter allows you to create a layout with resizable panes. The orientation of a splitter can be set to vertical or horizontal. Depending on the direction the drawer opens within the page, you can place a splitter at the start or end of a drawer.

Reviewer: Would we want to explicitly state where the splitter should be placed based on opening direction?

### Overlay drawer

An overlay drawer appears "on top" of page content, and must be minimized or closed in order for users to view the content that is covered by the expanded drawer. Overlay drawers in default and glass mode will have the `--pf-t--global--background--color--floating--default` token applied.

<div class="ws-docs-content-img">
![A drawer overlays page content, displaying a shadow where the primary page content is overlaid.](./img/drawer-overlay.svg)
</div>

### Inline drawer

> An inline drawer is placed beside page content, making the rest of the page content more compact (but still visible). All inline drawer variants have a `--pf-t--global--background--color--primary--default` token applied.

Reviewer 2: May be worth calling out that inline drawers in Glass will have no background color. Thoughts?

<div class="ws-docs-content-img">
![A drawer is beside page content, similar to a split screen view.](./img/drawer-inline.svg)
</div>

### Overlay pill

> A rounded overlay drawer is available across all themes. Apply the `pf-m-pill` modifier—either by applying the class or using the `isPill` prop for React—to enable the rounded style. The `--pf-t--global--border--radius--medium` (16px) border radius token and `--pf-t--global--border--color--subtle` border color token are applied to all rounded drawers. When placing a rounded drawer in a page, use the `global/spacer/inset/page-chrome` gutter token to ensure consistent spacing.

Reviewer: ```suggestion
A rounded overlay drawer is available across all themes. Apply the `pf-m-pill` class modifier or the `isPill` React property to enable the rounded style. The `--pf-t--global--border--radius--medium` (16px) border radius token and `--pf-t--global--border--color--subtle` border color token are applied to all rounded drawers. When placing a rounded drawer in a page, use the `global/spacer/inset/page-chrome` gutter token to ensure consistent spacing.
```
Just to remove some emdashes

The rounded overlay drawer has a `.pf-v6-u-box-shadow-md` (medium box shadow) applied to the entire drawer.

For an overlay pill drawer sitting above the main content area, the margin around the overlay drawer should use gutter spacer `--pf-t--global--spacer--gutter--default` on the top, bottom, and right if possible (or look visually equivalent if another code implementation is needed).

<div class="ws-docs-content-img">
![Example of an overlay pill drawer.](./img/Pill-Overlay.svg)
</div>

#### Compass layout - Pill overlay

> To see the drawer in a compass layout, view the [compass layout demo](/components/compass/org-demos/card-and-data-view-layout/).

Reviewer: Is it our recommendation that a pill drawer should be used for Compass layouts? If so, could we add that verbiage here?

Also, maybe "To see the pill overlay drawer..." instead

<div class="ws-docs-content-img">
![Example of an overlay pill drawer in a Compass layout.](./img/compass-overlay-drawer.svg)
</div>

#### Over main page content - Pill overlay

<div class="ws-docs-content-img">
![Example of an overlay pill drawer over main page content.](./img/Pill-Overlay-over-main-page.svg)
</div>

#### In main page area - Pill overlay

<div class="ws-docs-content-img">
![Example of an overlay pill drawer in the main page area.](./img/Pill-Overlay-In-main-page.svg)

> </div>

Reviewer: Will these sections have more content added? Just wasnt sure about only having images for these sections and no description/recommendations supplementing them.

Author: I didnt intend to add more to these section at the moment. I was trying to display all the drawer layout options that could arise. I can add quick descriptions under each if that makes more sense. @thatblindgeye

### Inline pill

> The rounded inline drawer shares the same base styling as the rounded overlay drawer. In the glass theme, a `.pf-v6-u-box-shadow-md` (medium box shadow) is applied to the entire drawer.

Reviewer: ```suggestion
The inline pill drawer shares the same base styling as the overlay pill drawer. In the glass theme, a `.pf-v6-u-box-shadow-md` (medium box shadow) is applied to the entire drawer.
```

Nit, just to use the pill verbiage vs "rounded"

<div class="ws-docs-content-img">
![Example of an inline pill drawer.](./img/Pill-Inline.svg)
</div>

#### Without masthead - Inline pill

<div class="ws-docs-content-img">
![Example of an inline pill drawer without a masthead.](./img/Pill-Inline-without-masthead.svg)

> </div>

Reviewer: Similar comment here as above about the section only containing an image
